controlplane/src/core/repositories/FederatedGraphRepository.ts

```diff
@@ -782,4 +788,188 @@ export class FederatedGraphRepository {
         }) as GraphApiKeyDTO,
     );
   }
+
+  public async registerClient(userId: string, federatedGraphName: string, clientName: string): Promise<string> {
+    if (!clientName) {
+      throw new Error('client name is empty');
+    }
+    const graph = await this.db.query.targets.findFirst({
+      columns: {},
+      with: {
+        federatedGraph: {
+          columns: { id: true },
+        },
+      },
+      where: and(
+        eq(schema.targets.name, federatedGraphName),
+        eq(schema.targets.organizationId, this.organizationId),
+        eq(schema.targets.type, 'federated'),
+      ),
+    });
+    if (graph === undefined) {
+      throw new Error(`could not find graph ${federatedGraphName}`);
+    }
+    const updatedAt = new Date();
+    await this.db
+      .insert(federatedGraphClients)
+      .values({
+        federatedGraphId: graph.federatedGraph.id,
+        name: clientName,
+        updatedAt,
+        createdBy: userId,
+        updatedBy: userId,
+      })
+      .onConflictDoUpdate({ target: [federatedGraphClients.federatedGraphId, federatedGraphClients.name], set: { updatedAt, updatedBy: userId } });
+
+    // To avoid depending on postgres, we do a second query to get the inserted client
```

Comment:
What do you mean by avoid postgres?

Reply:
MySQL doesn't support returning the ID from the insert, it's PG/SQLite only.